### DISCUSSION TITLE
PUT, PATCH, and DELETE requests

### DISCUSSION THREAD

**Discussion by howbizarre** [nuxt/nuxt#19065] [Questions]
Guys,

I am using the PUT and PATCH requests, but Nuxt type does not support them (`(property) method?: "get" | "GET" | "POST" | "post" | globalThis.Ref<"get" | "GET" | "POST" | "post" | undefined> | undefined`)

The requests looks like `await useFetch("/api/fileName", { method: "PATCH", body: { some: data } });`

Interestingly, "/server/api/fileName.patch.ts" works as I expect in the development environment, but I'm wondering if I should build it for production this way or change it to the POST request - "/api/fileName.post.ts", with some additional checks.

For now, I silence the TS compiler with // @ts-ignore

What do you think?
I do not have direct access to the production.

**Comment by TheAlexLichter** [comment]
I think that's a bug. All HTTP methods should be allowed by default. ☺️
Feel free to raise an issue 👍🏻

**Comment by howbizarre** [comment]
@manniL done #19077
